schemas/EGA.ArrayAssay.json

```diff
@@ -0,0 +1,89 @@
+        "allOf": [
+          {
+            "$ref": "https://github.com/EbiEga/ega-metadata-schema/tree/main/schemas/EGA.common-definitions.json#/definitions/object_coreIDs"
+          },
+          {
+            "properties": {
+              "accession": {
+                "$ref": "https://github.com/EbiEga/ega-metadata-schema/tree/main/schemas/EGA.common-definitions.json#/definitions/EGA-ArrayAssay-id-pattern"
+              }
+            }
+          }
+        ]        
+      },
+
+      "arrayAssay_center": {
+        "type": "string",
+        "title": "Centername that performed the ArrayAssay",
+        "description": "The name of the center (e.g. 'EBI-TEST') responsible for the profiling by microarray (executed the ArrayAssay), if applicable (in case it’s different from the center submitting metadata)."
+      },
+
+      "arrayAssay_date": {
+        "type": "string",
+        "title": "Date of the ArrayAssay",
+        "description": "ISO date (format YYYY-MM-DD - e.g. '2021-05-15') when the ArrayAssay took place. If the protocols are long enough, the date shall be the day the ArrayAssay concluded.",
+        "$ref": "https://github.com/EbiEga/ega-metadata-schema/tree/main/schemas/EGA.common-definitions.json#/definitions/EGA-ISO-date-YYYY-MM-DD-pattern"
+      },
+
+      "sample_number": {
+        "type": "integer",
+        "title": "Number of samples of the ArrayAssay",
+        "description": "Number of samples included in the Assay (i.e. pooled into one single microarray, labeled differently). One sample corresponds to one biological replicate [EFO:0002091] (it could be the genetic content from a single cell, a tissue… from a single individual or from several individuals). Shall not be mistaken for technical replicates [CHEBI:24432] being used several times (see https://www.ebi.ac.uk/seqdb/confluence/pages/viewpage.action?spaceKey=EGA&title=Sample+Representation)."
+      },
+
+      "sample_Labels": {
+        "type": "array",
+        "title": "Array of Sample-label pairs of the ArrayAssay",
+        "description": "Sample-Label pairs (format being 'EGAN[0-9]{11}:Label' - e.g. 'EGAN00000000001:Cy3') to know which samples used in this assay are labeled by which chemicals. Could be omitted if the array is of one single label and colour. #! Using an empty item list but definying the sample-label pairs as possible additionalItems we create the correct constraint: anything but a sample-label pair is rejected, but we can add as many as we want.",
+        "additionalProperties": false,
+        "items": [],
+        "additionalItems": {
+          "type": "string",
+          "title": "Sample-label pair",
+          "pattern": "^EGAN\\d{11}:[a-zA-Z0-9 ]+$"
+        }
```

Comment:
That is a good question, and for now what I chose was to make ``sample_Labels`` optional (not required), allowing only the pattern ``EGAN...:LABEL``. We could indeed make use of the aliases, but in that case we would also need:

- Providing the additional term "**centername**" of samples, unless the centername of the assay was the same as the centername that submitted the samples. 
- Additional **validation** steps upon submission checking that the aliases exist within the given centername.

Without this I feel it's a bit risky allowing aliases in those fields. Nevertheless, it is true that we may want to allow them without validating in case we simply want the user to know that the given "alias" was labeled that way. 

It's a trade off between how easy it is to provide wrong metadata and how difficult we make the process for the user.

Reply:
But the additional validation step will be necessary anyway, won't it? Before creating the final objects, we will need to verify that the referenced sample accessions exist. Or did I misunderstand you?

Reply:
> But the additional validation step will be necessary anyway, won't it? Before creating the final objects, we will need to verify that the referenced sample accessions exist. Or did I misunderstand you?

There should be, but from the perspective of the schemas alone, that validation cannot take place. I simply thought that using EGAN IDs would be less prone to human mistakes than using centernames and aliases. Both should be eventually checked, but since that is out of the scope of these schemas...

Anyway, I can change the label nodes to use Aliases and Centernames, and just make sure that we validate them eventually. This way submissions could be done in one go, without having to wait for EGANs to be assigned to your samples.